nautilus-extension/src/menu_provider.rs

```diff
@@ -184,6 +184,28 @@ macro_rules! menu_provider_iface {
             top_menu.to_g_list(Box::into_raw(Box::new(files_vec)) as *mut c_void)
         }
 
+        #[no_mangle]
+        pub extern "C" fn $get_background_items_fn(_provider: *mut c_void, window: *mut GtkWidget, current_folder: *mut NautilusFileInfo) -> *mut GList {
+            if current_folder.is_null() {
+                return ptr::null_mut() as *mut GList;
+            }
+
+            let file_info = FileInfo::new(current_folder);
+
+            let file_items: Vec<MenuItem> =
+                match *$rust_provider.lock().unwrap() {
+                    Some(ref p) => p.get_background_items(window, &file_info),
+                    None => Vec::new(),
+                };
+
+            // dummy top-level Menu for easy recursion
+            let top_menu = Menu {
+                menu_items: file_items,
+            };
+
+            top_menu.to_g_list(Box::into_raw(Box::new(current_folder)) as *mut c_void)
```

Comment:
Thanks for working on this feature!

The crash is caused by this line. It should be:

    top_menu.to_g_list(Box::into_raw(Box::new(file_info)) as *mut c_void)

Reply:
Thank you!